transport/grpc/config.go

```diff
@@ -78,9 +80,43 @@ type TransportConfig struct {
 // inbounds:
 //   grpc:
 //     address: ":80"
+//     tls:
+//       enabled: true
+//       keyFile: "/path/to/key"
+//       certFile: "/path/to/cert"
 type InboundConfig struct {
 	// Address to listen on. This field is required.
-	Address string `config:"address,interpolate"`
+	Address string           `config:"address,interpolate"`
+	TLS     InboundTLSConfig `config:"tls,optional"`
+}
+
+func (c InboundConfig) inboundOptions() ([]InboundOption, error) {
+	return c.TLS.inboundOptions()
+}
+
+// InboundTLSConfig configures a gRPC inbound TLS credentials.
+type InboundTLSConfig struct {
+	Enabled  bool   `config:"enabled,optional"`
+	CertFile string `config:"certFile,optional"`
+	KeyFile  string `config:"keyFile,optional"`
+}
+
+func (c InboundTLSConfig) inboundOptions() ([]InboundOption, error) {
```

Comment:
nit since this exposed: does this need to return an array? Would the TLS Config ever return more than one `InboundOption`?
eg
```golang
func (c InboundTLSConfig) inboundOption() (InboundOption, error) {
```

Reply:
I think you are correct. I will confirm and update.

Reply:
This can provide 0 or 1 options, and this lays the plumbing for threading more if necessary.

Reply:
Actually, no.

Reply:
Yeah - after tinkering, I totally agree that it is better returning slice all the way through.